include/hdlConvertor/hdlObjects/hdlTypeDec.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include <hdlConvertor/hdlObjects/named.h>
+#include <hdlConvertor/hdlObjects/iHdlObj.h>
+
+
+namespace hdlConvertor {
+namespace hdlObjects {
+
+/*
+ * HDL AST node for module declaration
+ * (part with ports and params for Verilog, Entity for VHDL)
+ * */
+class HdlTypeDec: public WithNameAndDoc, public iHdlObj {
+public:
+	enum TypeClass {hdltc_error, hdltc_enum, hdltc_array, hdltc_struct, 
+		hdltc_physical, hdltc_integral, hdltc_float};
+	HdlTypeDec(const std::string& name);
```

Comment:
There is a little problem as the types in SV can be defined anonymously.

```c
stuct { int a; } b;
```
This can be easily solved by name="", however it may be also worth considering if we should put the HdlTypeDec always inside HdlVariableDef or not. It may be beneficial, but on the other way it may only complicate the things.

Reply:
The favorite solution I've seen in this case is to create a pattern for an "anonymous" type.  The patterns I've liked generate an identifier that isn't valid in the language to avoid collisions.  That way `b` still has a named type, though it isn't a name that can be used anywhere else.

Reply:
W.r.t. `HdlVariableDef`, I agree that there is similarity in a "name" and "type-ish" pair.  But in VHDL, at least, a type declaration is very different in syntax and semantics from the subtype indication of a variable/constant/signal/port/generic declaration.  So I think it's not a good fit.

Reply:
> an identifier that isn't valid in the language to avoid collisions

Maybe we can just add a flag which signalize if the  type is anonymous or not.

> So I think it's not a good fit.

ok, makes sence

Reply:
We could add a flag and perhaps leave the name blank.  But I like having something in error messages.  For example, I've seen a compiler error message something like, "type INTEGER does not match type (error)".  I liked this because I knew that one side was happy, but the other side was missing a data type.  So I'm imagining something like "(anonymous)" for the type name and maybe a flag, too.  I think the flag is unnecessary, though, since users can't define a data type name "(anonymous)", so there won't be any chance of trying to resolve this name and accidentally getting the anonymous type.  Giving names to exceptional types like these (error and anonymous) helps with error messages and with AST viewing/debugging.

What do you think?

Reply:
OK, lets have a special name for anonymous types, maybe we should also put some code location info in to generated type name. We do not need a flag for anonymous types, if we have a method which can check if type is anonymous it is sufficient.